internal/constants/constants.go

```diff
@@ -444,6 +444,9 @@ const DefaultBranchName = "main"
 // UnstableConfig is the config key used to determine whether the user has opted in to unstable commands
 const UnstableConfig = "optin.unstable"
 
+// AsyncRuntimeConfig is the config key used to determine whether the user has opted in to async runtimes
+const AsyncRuntimeConfig = "optin.unstable.async.runtime"
```

Comment:
```suggestion
const AsyncRuntimeConfig = "optin.unstable.async_runtime"
```

Reply:
We have rules to enforce config key structure and it flags the underscore in the key as invalid. Should we change that for this story?

Reply:
hmmm, the rule is to avoid that we introduce something like `optin_unstable_async_runtime`, which would be inconsistent. But to me the dots are for grouping, so for example anything under `optin.unstable` is obviously about toggling the unstable setting for that given property. With that in mind `optin.unstable.async` doesn't really make sense as a "group" to me since we're unlikely to put anything else there, which is why I suggested `async_runtime`.

With that context out of the way; maybe we can update the rule to require at least one dot but to otherwise accept underscores? We'll have to guard against monstrosities like `optin_unstable.async_runtime` in our code reviews.. :p